src/components/Editor/components/TextEditor.js

```diff
@@ -32,14 +31,12 @@ class TextEditor extends React.Component {
     window.addEventListener("close", this.onLeave);
   }
 
-  componentWillUnmount = () => {
-    //this.checkDirty();
-    window.removeEventListener("beforeunload", this.onLeave);
-    window.removeEventListener("close", this.onLeave);
-  };
+  componentWillUpdate() {}
+
+  componentWillUnmount = () => {};
```

Comment:
Will fix/update. Makes more sense to keep it in the programsReducer, will move it there. Aseem and I were pondering how to show if a specific file is dirty too, since right now dirty prop is true if any file has been edited/unsaved regardless if it is currently displayed in the text editor--will ask you next week.